### PR TITLE
[DOCS-201] Update API Methodologies

### PR DESCRIPTION
**Pull request by mjang-cobalt**
## Changelog

Followed same pattern as discussed with CSM. Based additional content on internal Confluence article.

### Added

n/a

### Updated

| Page | Deploy Preview | Comment |
| ----- | ----- | ----- |
| [API Methodologies] | [[Link](https://deploy-preview-234--cobalt-docs.netlify.app/platform-deep-dive/pentests/pentest-process/methodologies/api-methodologies/)] | [Comment] |

## Preview This Change

To see how this change looks in production, scroll down to **Deploy Preview**. Select the link that looks like `https://deploy-preview-<num>--cobalt-docs.netlify.app/`

## Variables

Help us support a “Write once, publish everywhere” single source of truth. If you see a line that looks like:

`{{% asset-categories %}}`

You’ve found a [shortcode](https://gohugo.io/content-management/shortcodes/) that we include in multiple documents.

You’ll find the content of the shortcode in the following directory:

https://github.com/cobalthq/cobalt-product-public-docs/tree/main/layouts/shortcodes

That shortcode has the same base name as what you see in the PR, such as `asset-categories.html`.

## Checklist for PR Author

[ ] Did you check for broken links and alt text?

Be sure to check for broken links and Alt Text issues. We have a partially automated process,
as described in this section of our repository README:
[Test Links and Alt Attributes](https://github.com/cobalthq/cobalt-product-public-docs/blob/main/README.md#test-links-and-alt-attributes).

### CODE REVIEW

File: content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md [hunk 1]
```diff
@@ -35,19 +35,191 @@ Learn more about these objectives from OWASP:
...
+- testssl.sh
+- [Dirble](https://github.com/nccgroup/dirble)
+- OWASP Zed Attack Proxy (ZAP)
+
+{{% /expand %}}
+
+{{% expand "Business and application logic mapping" %}}
+</br>
+
+Our pentesters, manually examine the target application to map its:
```
Comment (ana-dashuk-cobalt) [inline content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md:73]:
```suggestion
Our pentesters manually examine the target application to map its:
```

File: content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md [hunk 2]
```diff
@@ -35,19 +35,191 @@ Learn more about these objectives from OWASP:
...
+
+- Business functions
+- Workflows
+- The underlying processes
+
+Our pentesters also build a matrix of access controls within the app based on
+
+- A list of available roles
+  - The actions supported for each role
```
Comment (ana-dashuk-cobalt) [inline content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md:82]:
```suggestion
Our pentesters also build a matrix of access controls within the app based on:

- A list of available roles
- The actions supported for each role
```

Is this the right formatting?

File: content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md [hunk 3]
```diff
@@ -35,19 +35,191 @@ Learn more about these objectives from OWASP:
...
+- Burp Suite Community/Professional Edition
+- OWASP Zed Attack Proxy (ZAP)
+
+{{% /expand %}}
+
+{{% expand "Automated web crawling and web scanner configuration tweaking" %}}
+</br>
+
+Our pentesters may use commercial and open source security tools to assess
+your API. They ensure that approporiate tools cover the whole scope of the app.
```
Comment (ana-dashuk-cobalt) [inline content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md:100]:
```suggestion
your API. They ensure that appropriate tools cover the whole scope of the app.
```

File: content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md [hunk 4]
```diff
@@ -35,19 +35,191 @@ Learn more about these objectives from OWASP:
...
+They also make sure to cover every segment for security issues.
+
+They may have to manually tweak their tools to ensure optimized performance. 
+
+In addition, our pentesters perform automated crawls to identify:
+
+- Pages available to unauthenticated users
+- The directory tree of yoru website
```
Comment (ana-dashuk-cobalt) [inline content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md:108]:
```suggestion
- The directory tree of your website
```

File: content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md [hunk 5]
```diff
@@ -35,19 +35,191 @@ Learn more about these objectives from OWASP:
...
+
+{{% /expand %}}
+
+{{% expand "Authenticated vulnerability scanning and manual crawling" %}}
+</br>
+
+Our pentesters run several tests in this area, including:
+
+- Automated crawling tests, followed by manual verification
+- Additional manual crawling tests for better coverage
```
Comment (ana-dashuk-cobalt) [inline content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md:125]:
```suggestion
- Additional manual crawling tests for better coverage:
```

File: content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md [hunk 6]
```diff
@@ -35,19 +35,191 @@ Learn more about these objectives from OWASP:
...
+
+{{% expand "Manual API vulnerability testing and exploitation" %}}
+</br>
+
+Cobalt pentesters use manual testing tools to identify and analyze the following aspects of the API asset:
+
+- Functionality
+- Workflows
+- Business logic
+- Vulnerabilities in deployment and implementationo
```
Comment (ana-dashuk-cobalt) [inline content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md:148]:
```suggestion
- Vulnerabilities in deployment and implementation
```

File: content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md [hunk 7]
```diff
@@ -35,19 +35,191 @@ Learn more about these objectives from OWASP:
...
+The assessment identifies known vulnerabilities, including those listed in the:
+
+- [OWASP API Top 10](https://owasp.org/www-project-api-security/)
+- [CVE](https://cve.mitre.org/)
+
+Our pentesters also:
+
+- Run injection attacks that probe the robustness of server-validation routines
+- Look for session management flaws that may allow user impersonation
+- Investigate flaws in access control that expose data or enable users to gain elevated privileges.
```
Comment (ana-dashuk-cobalt) [inline content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md:159]:
```suggestion
- Investigate flaws in access control that expose data or enable users to gain elevated privileges
```

File: content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md [hunk 8]
```diff
@@ -35,19 +35,191 @@ Learn more about these objectives from OWASP:
...
+
+Beyond these tests, our pentesters also:
+
+- Test how well the design and implementation protects data against unauthorized access or disclosure
+- Review how the endpoints validate input
+- Review how the API handles access tokens
+- Review how the API responds to error conditions or invalid states
+- Consider how resistant the API is to accidental misuse or unintended mistakes by a user that would lead to security issues
+
+For microservices, our pentesters focus on the interactions between different systems. Specifically, we examine
```
Comment (ana-dashuk-cobalt) [inline content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md:169]:
```suggestion
For microservices, our pentesters focus on the interactions between different systems. Specifically, we examine:
```

File: content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md [hunk 9]
```diff
@@ -35,19 +35,191 @@ Learn more about these objectives from OWASP:
...
+{{% expand "Ongoing assessments" %}}
+</br>
+
+Our pentesters report their findings, in real time, through the Cobalt platform.
+They also:
+
+- Assess all risks
+- Recommend steps for remediation
+
+You're welcome to communicate with our pentesters for each of their findings.
```
Comment (ana-dashuk-cobalt) [inline content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md:204]:
```suggestion
Our pentesters report their findings, in real time, through the Cobalt platform.
They also:

- Assess all risks
- Recommend steps for remediation

You're welcome to communicate with our pentesters for each of their findings.
```

This is another potential shortcode.

File: content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md [hunk 10]
```diff
@@ -35,19 +35,191 @@ Learn more about these objectives from OWASP:
...
+Our pentesters report and triage all vulnerabilities during the assessment. You
+can review details of all findings, in real time, through the Cobalt platform.
+In these findings, as well as in any report, our pentesters include detailed
+information on how you can:
+
+- Remediate each finding
+- Improve your overall security posture
+
+If you remediate findings during the supported pentest period, our pentesters
+can retest your updated components against each discovered vulnerability.
```
Comment (ana-dashuk-cobalt) [inline content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md:220]:
```suggestion
Our pentesters report and triage all vulnerabilities during the assessment. You
can review details of all findings, in real time, through the Cobalt platform.
In these findings, as well as in any report, our pentesters include detailed
information on how you can:

- Remediate each finding
- Improve your overall security posture

If you remediate findings during the supported pentest period, our pentesters
can retest your updated components against each discovered vulnerability.
```

Another candidate for a shortcode.

**Review by ana-dashuk-cobalt — APPROVED** [review]
@mjang-cobalt Approving with several comments. We may need to include more shortcodes for all pentest methodologies.

File: content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md [hunk 11]
```diff
@@ -35,19 +35,191 @@ Learn more about these objectives from OWASP:
...
+Our pentesters also confirm that they can access and run documented commands on
+your API. If you require access tokens or API keys to use your API, our
+pentesters may need accounts on your system, with instructions, to set that up.
+
+Our pentesters may use scanning tools, including:
+
+- Burp Suite Community/Professional Edition
+- testssl.sh
+- [Dirble](https://github.com/nccgroup/dirble)
+- OWASP Zed Attack Proxy (ZAP)
```
[the remark this answers is not in the document]
Comment (mjang-cobalt) [reply content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md:66]: Hmm, OK, I missed the duplicate

File: content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md (hunk 9 above)
Comment (mjang-cobalt) [reply content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md:204]: Hmm... did we release where pentesters report their findings through the platform?

**Comment by mjang-cobalt** [comment]
Thanks for your patience with this, @ana-dashuk-cobalt . I've incorporated short codes where it made sense. 

Lesson -- besides always checking with vale, don't overdo it with shortcodes on other files. I had rebase/conflict issues.

File: content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md (hunk 9 above)
Comment (ana-dashuk-cobalt) [reply content/en/Platform Deep Dive/Pentests/Pentest Process/Methodologies/api-methodologies.md:185]: @mjang-cobalt No, we don't have a section describing findings yet. Once we have it, we'll add a cross-reference. We can at least mention that they can view findings on the **Findings** tab of the pentest page.

**Pull request merged**